packages/manager/src/features/components/PlansPanel/utils.ts

```diff
@@ -337,11 +345,24 @@ export const extractPlansInformation = ({
 }: ExtractPlansInformationProps) => {
   const plansForThisLinodeTypeClass: PlanWithAvailability[] = plans.map(
     (plan) => {
+      const isMtcTt2025Plan = isMtcTT2025Plan(plan);
+
+      // Disable all 512GB plans apart from GPUs and mtc-tt-2025
```

Comment:
With the addition, we should just make this clearer
```suggestion
      // Special handling for 512GB plans:
      // - Generally disabled when `disableLargestGbPlansFlag` is true
      // - Exceptions: GPU plans and MTC_TT plans
```